database_change_log/EIA-101_update_dashboard_data_function.sql

```diff
@@ -0,0 +1,163 @@
+	or
+		(ead.user_ref ilike ' || quote_literal(query_string)  || ')
+	)
+	union
+	select
+	app.application_start_date as "createdDate",
+	ead.unique_app_id,
+	ats."applicationType",
+	ead.doc_count,
+	ead.payment_amount,
+	ead.user_ref,
+	ead.main_postcode as main_postcode,
+	' || result_count || ' as result_count
+	from "Application" app inner join
+	"ExportedApplicationData" ead
+	on app.application_id = ead.application_id
+	inner join
+	"ApplicationTypes" ats
+	on app."serviceType" = ats.id
+	where app.user_id=$1
+	and (
+		(ead.unique_app_id ilike ' || quote_literal(query_string)  || ')
+	or
+		(ats."applicationType" ilike ' || quote_literal(query_string)  || ')
+	or
+		(ead.user_ref ilike ' || quote_literal(query_string)  || ')
+	)
+	order by ' ||  _orderby || ' ' || _direction || '
+	LIMIT $2 OFFSET $3'
+USING _user_id, _limit, _offset;
+ELSE
+select count(*) into result_count
+	from "Application" app inner join
+	"ExportedEAppData" ead
+	on app.application_id = ead.application_id
+	inner join
+	"ApplicationTypes" ats
+	on app."serviceType" = ats.id
+	where app.user_id=_user_id
+
+	and (
+		(ead.unique_app_id ilike query_string)
+	or
+		(ead.user_ref ilike query_string)
+	or
+		(ats."applicationType" ilike query_string)
+	);
+
+RETURN QUERY EXECUTE '
+	select
+	app.application_start_date as "createdDate",
+	ead.unique_app_id,
+	ats."applicationType",
+	ead.doc_count,
+	ead.payment_amount,
+	ead.user_ref,
+	'''' as main_postocde,
+	' || result_count || ' as result_count
+	from "Application" app inner join
+	"ExportedEAppData" ead
+	on app.application_id = ead.application_id
+	inner join
+	"ApplicationTypes" ats
+	on app."serviceType" = ats.id
+	where app.user_id=$1
+	and (
+		(ead.unique_app_id ilike ' || quote_literal(query_string)  || ')
+	or
+		(ats."applicationType" ilike ' || quote_literal(query_string)  || ')
+	or
+		(ead.user_ref ilike ' || quote_literal(query_string)  || ')
+	)
+	union
+	select
+	app.application_start_date as "createdDate",
+	ead.unique_app_id,
+	ats."applicationType",
+	ead.doc_count,
+	ead.payment_amount,
+	ead.user_ref,
+	ead.main_postcode as main_postcode,
+	' || result_count || ' as result_count
+	from "Application" app inner join
+	"ExportedApplicationData" ead
+	on app.application_id = ead.application_id
+	inner join
+	"ApplicationTypes" ats
+	on app."serviceType" = ats.id
+	where app.user_id=$1
+	and (
+		(ead.unique_app_id ilike ' || quote_literal(query_string)  || ')
+	or
+		(ats."applicationType" ilike ' || quote_literal(query_string)  || ')
+	or
+		(ead.user_ref ilike ' || quote_literal(query_string)  || ')
+	)
+	order by ' ||  _orderby || ' ' || _direction || ',' || _secondaryorderby || ' ' || _secondarydirection || '
+	LIMIT $2 OFFSET $3'
+USING _user_id, _limit, _offset;
+ END IF;
+END;
+$BODY$
+  LANGUAGE plpgsql VOLATILE
+  COST 100
+  ROWS 1000;
+ALTER FUNCTION public.dashboard_data_eapp(integer, integer, integer, text, text, text, text, text)
+  OWNER TO postgres;
```

Comment:
It think it will be well worth formatting this SQL a bit more for readability. Mostly, indention of the string literal queries. I avoid rigidly following a specific style but I usually go with something like the example below. It keeps lines short and avoids too much nesting. **Any style focusing on clarity, readability, and easy maintenance is good in my book.**
```
SELECT
    table_1.column_a,
    table_1.column_b,
    table_2.column_c
FROM
    table_1
INNER JOIN
    table_2
    ON table_1.column_a = table_2.column_a
WHERE
    table_1.column_b = 'criteria'
    AND table_2.column_c = 'criteria'
```

Reply:
Good shout, I'm not much of an expert with sql files so I didn't know if there were any formatting standards, but I'll look around for a VS Code plugin. I'm sure one exists.